### ISSUE TITLE
How do we add Ingest Budget to this docker? 

### ISSUE THREAD

**Issue by mhobotpplnet** [SumoLogic/sumologic-collector-docker#83]
Just trying to work with some new fields , and was wondering where do I add _budget field ? is it the sources or collector it self?

**Comment by maimaisie** [comment]
Hi @mhobotpplnet , you can supply your own user.properties file to specify the budget field on the collector. 

This [section](https://github.com/SumoLogic/sumologic-collector-docker#configure-collector-in-userproperties-file) in the README explains how to supply your own properties file when starting the container. 

This [doc page](https://help.sumologic.com/03Send-Data/Installed-Collectors/05Reference-Information-for-Collector-Installation/06user.properties) talks more in detail on what fields are supported in this file. Search for "Ingest Budget" on this page to see how to format this field in user.properties file. It should be something like `fields=_budget=<your_budget_field_value>`

**Comment by mhobotpplnet** [comment]
@maimaisie yeah I saw that, I got bit of an issue where I supply the secrets.

So my question is, if I have say this in my docker-compose, and if I add custom user.properties set as well, will info below also be added? Or do I need to hardcode info below in custom user.properties file? 
```
      SUMO_CLOBBER: 'false'
      SUMO_COLLECTOR_NAME_PREFIX: "${MYSTUFF_ENVIRONMENT}-"
      SUMO_SOURCES_JSON: /config/sumo/sources.json
      SUMO_SYNC_SOURCES: 'false'
```

**Comment by mhobotpplnet** [comment]
Example how it looks w/o custom properties atm 

```
root@sumologic-collector-docker-sumologic-12:/# cat /opt/SumoCollector/config/us
er.properties
#
#Tue May 21 16:48:12 UTC 2019
accessid=myID
name=dev-sumologic-collector-docker-sumologic-12
sources=/config/sumo/sources.json
url=https\://collectors.sumologic.com
ephemeral=true
clobber=false
```

**Comment by maimaisie** [comment]
Unfortunately it doesn't support combining custom user.properties file and properties passed from environment variables. So all the properties should be specified through user.properties if you supply your own.

In the future we will add ingest budget assignment through environment variable as well.

**Comment by mhobotpplnet** [comment]
Show me where you would expose that as an env, I might code it and push it as PR.

**Comment by mhobotpplnet** [comment]
Probably somewhere here right? 

https://github.com/SumoLogic/sumologic-collector-docker/blob/70ca21adb4957ab3e189ddb57c40e044903e92d7/run.sh

**Comment by maimaisie** [comment]
Yes, basically similar to this: https://github.com/SumoLogic/sumologic-collector-docker/pull/79

**Comment by mhobotpplnet** [comment]
Very odd, so I added 
```SUMO_INGEST_BUDGET=${SUMO_INGEST_BUDGET:_budget=test-budget}``` at the top of the file and for declared options I added ```["SUMO_INGEST_BUDGET"]="fields"``` ran the docker locally and I can't see any user.properties changes. So I did with docker run -e SUMO_INGEST_BUDGET thinking I must add this env then, that did not do anything else.  

Any other pointers? 

My user.properties file should at the end have
```fields=_budget=test-budget```

**Comment by maimaisie** [comment]
> Update: If I run docker -e for SUMO_COLLECTOR_EPHEMERAL=false , it works just fine which kind of tells me that the sumo wrapper has those variables predefined.

That is very strange. They should be independent properties. Can you maybe create a PR with what you had so I can review and possibly find where the problem is?

**Comment by MattPumphrey** [comment]
This would be a great feature to have, as if we are running an entirely dynamic workload that the nodes scale but up and down and the only agent on them is the sumologic collector for log collection, then having a seperate user.properties file when we have been using the generated one does not bode with our ability to use the ingest budgets on scale up of nodes.

I just started looking at this and if I find a workable solution, I will make a pr.
